### PR TITLE
enh(conf/broker) add bbdo_version in broker configuration

### PR DESCRIPTION
**Pull request by a-launois**
## Description

add bbdo_version setting in broker configuration, to handle bbdo_version V3

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/install/createTables.sql [hunk 1]
```diff
@@ -452,6 +452,7 @@ CREATE TABLE `cfg_centreonbroker` (
   `stats_activate` enum('0','1') DEFAULT '1',
   `daemon` TINYINT(1),
   `pool_size` int(11) DEFAULT NULL,
+  `bbdo_verson` varchar(50) DEFAULT '3.0.0',
```
Comment (a-launois) [inline www/install/createTables.sql:455]:
```suggestion
  `bbdo_version` varchar(50) DEFAULT '3.0.0',
```

**Review by callapa — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

File: www/include/configuration/configCentreonBroker/formCentreonBroker.php [hunk 2]
```diff
@@ -58,6 +58,40 @@
...
+    $data['filename'] = html_entity_decode($data['filename']);
+    $data['cache_directory'] = html_entity_decode($data['cache_directory']);
+    $data['log_directory'] = html_entity_decode($data['log_directory']);
+    $data['log_filename'] = html_entity_decode($data['log_filename']);
+    $data['bbdo_version'] = html_entity_decode($data['bbdo_version']);
+    $data['command_file'] = html_entity_decode($data['command_file']);
+
+    return $data;
+}
+
```
Comment (TamazC) [inline www/include/configuration/configCentreonBroker/formCentreonBroker.php:94]:
```suggestion
/**
 * @param array<string,mixed> $data
 * @return array<string,mixed>
 */
function htmlEncodeBrokerInformation(array $data): array
{
    return array_map('encodeToHTMLEntities', $data);
}

/**
 * @param array<string,mixed> $data
 * @return array<string,mixed>
 */
function htmlDecodeBrokerInformation(array $data): array
{
    return array_map('decodeHTMLEntity', $data);
}

function encodeToHTMLEntities(string $text): string
{
    return htmlentities($text);
}

function decodeHTMLEntity(string $htmlEntity): string
{
    return html_entity_decode($htmlEntity);
}

```

is more compact and elegant I think. However missing array keys would not be initialized.

File: www/class/centreonConfigCentreonBroker.php [hunk 3]
```diff
@@ -684,38 +684,86 @@ public function insertConfig($values)
     {
         $objMain = new CentreonMainCfg();
```
Comment (TamazC) [inline www/class/centreonConfigCentreonBroker.php:685]: I think the function is too big and it does more than one thing : creates records into "cfg_centreonbroker" table and into "cfg_centreonbroker_log" table.
Sometimes it has no returns and thrown exceptions aren't handled (maybe they should be logged ?)
I think we should split it in 3 parts and make them more simple.

File: www/class/centreonConfigCentreonBroker.php [hunk 4]
```diff
@@ -684,38 +684,86 @@ public function insertConfig($values)
...
+        }
+        if (isset($values['stats_activate']['stats_activate'])) {
+            $columnNames[] = 'stats_activate';
+        }
+        if (isset($values['log_max_size'])) {
+            $columnNames[] = 'log_max_size';
+        }
+        if (isset($values['bbdo_version'])) {
+            $columnNames[] = 'bbdo_version';
+        }
```
Comment (TamazC) [inline www/class/centreonConfigCentreonBroker.php:714]: The logic here is copied in lines 743-765.

**Review by callapa — DISMISSED** [review]

File: www/class/centreonConfigCentreonBroker.php [hunk 5]
```diff
@@ -674,48 +674,107 @@ public function getTypeId($typeName)
...
+    }
+
     /**
      * Insert a configuration into the database
      *
      * @param array $values The post array
      * @return bool
      */
-    public function insertConfig($values)
+    public function insertConfig($values): bool
```
Comment (a-launois) [inline www/class/centreonConfigCentreonBroker.php:721]:
```suggestion
    public function insertConfig(array $values): bool
```

**Review by callapa — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by callapa — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**